app/src/forms/auth/middleware/userAccess.js

```diff
@@ -83,39 +56,31 @@ const _getForm = async (currentUser, formId, includeDeleted) => {
   return form;
 };
 
-/**
- * Express middleware that adds the user information as the res.currentUser
- * attribute so that all downstream middleware and business logic can use it.
- *
- * This will fall through if everything is OK. If the Bearer auth is not valid,
- * this will produce a 403 error.
- *
- * @param {*} req the Express object representing the HTTP request.
- * @param {*} _res the Express object representing the HTTP response - unused.
- * @param {*} next the Express chaining function.
- */
-const currentUser = async (req, _res, next) => {
+const setUser = async (req, _res, next) => {
   try {
-    // Validate bearer tokens before anything else - failure means no access.
-    const bearerToken = _getBearerToken(req);
-    if (bearerToken) {
-      const ok = await keycloak.grantManager.validateAccessToken(bearerToken);
-      if (!ok) {
-        throw new Problem(403, { detail: 'Authorization token is invalid.' });
-      }
-    }
-
-    // Add the request element that contains the current user's parsed info. It
-    // is ok if the access token isn't defined: then we'll have a public user.
-    const accessToken = _getAccessToken(req);
-    req.currentUser = await service.login(accessToken);
-
+    const token = await jwtService.getTokenPayload(req);
+    req.currentUser = await service.login(token);
     next();
   } catch (error) {
     next(error);
   }
 };
 
+const currentUser = async (req, res, next) => {
```

Comment:
It looks like some of the changes in PR1263 were lost on the merge. The only important thing is to wrap the code in `currentUser` (and all middleware) in `try`/`catch` blocks so that requests won't hang on error. Something like:
```javascript
const middleware = async (req, _res, next) => {
  try {
    // If possible the helper functions should throw a Problem rather than return a
    // response that needs handling logic.
    const formId = _do_stuff(req);
    const foo = service.getForm(formId);

    next();
  } catch (error) {
    next(error);
  }
};
```

Reply:
great catch and fixed this up (and related test)

Reply:
Looks good, the request is no longer hanging when the token expires.